data/json/npcs/exodii/exodii_merchant_missions.json

```diff
@@ -52,11 +52,35 @@
       "failure": "Fine an' fine.  Come back to this'n if you like to try again."
     }
   },
+  {
+    "id": "EXODII_WAREHOUSE",
+    "name": { "str": "Find the lost Exodii warehouse and reboot its locator antenna." },
+    "type": "mission_definition",
+    "goal": "MGOAL_GO_TO_TYPE",
+    "destination": "exo_warehouse_1",
+    "end": { "effect": [ { "u_add_var": "u_found_warehouse", "type": "mission", "context": "completed", "value": "yes" } ] },
+    "difficulty": 5,
+    "value": 2,
+    "item": "warehouse_rebooter",
+    "origins": [ "ORIGIN_SECONDARY" ],
+    "dialogue": {
+      "describe": "",
+      "offer": "",
+      "accepted": "King's own luck!",
+      "rejected": "That be a choice an' you can tass.  Come back to this'n if you like.",
+      "advice": "What ye'r searching is safehouse-like, sand, stone, metal an' all.",
+      "inquire": "Us'n waits patient-like for that'.",
+      "success": "Fine an' fine.  We can pass it in when you like.",
+      "success_lie": "Fine an' fine.  We can pass it in when you like.",
+      "failure": "Fine an' fine.  Come back to this'n if you like to try again."
+    }
+  },
   {
     "id": "TALK_EXODII_MERCHANT_mission_1_intro",
     "type": "talk_topic",
     "dynamic_line": "Them as offer as can, an it be so. Aye, there's summat the old Scythy been larkin' up Rubik's tailpipe.  B'ain't tass right for a clanker like this'n, but a pop-an-work like you'n, maybe well and suited.",
     "//~": "Just because you asked, I think we can work something out.  Yes, the Scythean has been bugging me about something.  It's not something a cyborg like me can do, but a mercenary would probably fit the bill.",
+    "run_eocs": { "id": "EOC_MISSION_EXODII_HUB_DRONE_CHECK" },
```

Comment:
It doesn't look like you can put "run_eocs" here it needs to be in "responses" or in "speaker_effect" looking at existing usage in talk_topic s (though I've no idea what the latter does)

Reply:
I'll try to see if there's a way to run that EoC every time you talk to Rubik without putting it everywhere. 

Checking for interactions with Rubik is a sure way to ensure both your presence and you being correlated to the drone's arrivale

Reply:
The logic was reworked as suggested on discord. It now checks if you ever interacted with hub instead